docs/mp/guides/migration_3.x.adoc

```diff
@@ -0,0 +1,80 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2022 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon MP 3.x Upgrade Guide
+:description: Helidon MP 3.x Upgrade Guide
+:keywords: helidon, porting, migration, upgrade, incompatibilities, 3.x
+:rootdir: {docdir}/../..
+
+include::{rootdir}/includes/mp.adoc[]
+
+
+In Helidon 3.x we have made some changes to APIs and runtime behavior. This guide
+will help you migrate a Helidon MP 2.x application to 3.x.
+
+
+== Java 17 Runtime
+
+Java 11 is no longer supported in Helidon 3. Java 17 or newer is required.
+
+
+== javax.* namespace to jakarta.* namespace
+
+Due to trademark restrictions imposed on the javax.* namespace, Jakarta EE will rename all of its specifications to use jakarta.* in order to move forward with evolving the platform with features as the cloud-native Java platform.
+In version 3.x Helidon supports MicroProfile 5 specification, which now is fully migrated to `jakarta.x` namespace.
+
+As the `javax` module is no longer in dependency management of Helidon parent pom files.
```

Comment:
Maybe add "a result" after the initial "As"?

Reply:
fixed